Brick-Studios/BrickEngine
# Added JSON writing and FloatingPointComparer

# Description
This PR adds writing to the json file and a fix for applying a new scene to the secondary layer without a game state change.

This also introduced a FloatingPointComparer which should be used when comparing floats.

# How can this be tested?

See [the PR in BeastArena](https://github.com/Brick-Studios/BeastArena/pull/48).

# Checklist:
